pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -719,13 +719,22 @@ func TestCIDRSetv6(t *testing.T) {
 			continue
 		}
 		if !tc.expectErr {
-			if p.String() != tc.expectedCIDR {
+			if p != nil && p.String() != tc.expectedCIDR {
 				t.Fatalf("unexpected allocated cidr: %s for %v", p.String(), tc.description)
 			}
 		}
+
 		p2, err := a.AllocateNext()
+		if err == nil && tc.expectErr {
+			t.Errorf("expected error but got none for %v", tc.description)
```

Comment:
can we convert this to t.Run and not print out the description manually

```
for _, tc := range cases {
  t.Run(tc.description, func(t *testing.T) {
    ...
    t.Errorf("a.AllocateNext() = nil, want error")
    ...
    t.Errorf("a.AllocateNext() = %v", want nil", err) // etc..
    ...
  })
```

Reply:
@bowei Then I should remove the description field itself from test cases. Because for below case I can write like

`p != nil && p.String() != tc.expectedCIDR`
I can use 
`t.Errorf("a.AllocateNext() =%+v, and want = %+v ", p.String(), tc.expectedCIDR)`